src/main/java/com/couchbase/lite/router/Router.java

```diff
@@ -1595,91 +1609,141 @@ private void sendContinuousChange(RevisionInternal rev) {
                 try {
                     os.write(json);
                     os.flush();
-                } catch (Exception e) {
-                    Log.e(TAG, "IOException writing to internal streams", e);
+                } catch (IOException e) {
+                    Log.e(TAG, "Exception writing to internal streams", e);
                 }
             }
         } catch (Exception e) {
             Log.w("Unable to serialize change to JSON", e);
         }
     }
 
+    private void sendContinuousChangeLastSequenceAndFinish(long lastSeq) {
+        if (callbackBlock != null) {
+            Map<String, Object> body = new HashMap<String, Object>();
+            body.put("last_seq", lastSeq);
+
+            String bodyString = null;
+            try {
+                bodyString = Manager.getObjectMapper().writeValueAsString(body) + '\n';
+            } catch (JsonProcessingException e) {
+                Log.w("Unable to serialize change to JSON", e);
+                return;
+            }
+
+            OutputStream os = connection.getResponseOutputStream();
+            try {
+                os.write(bodyString.getBytes());
+                os.flush();
+            } catch (IOException e) {
+                Log.e(TAG, "Exception writing to internal streams", e);
+            } finally {
+                try {
+                    if (os != null)
+                        os.close();
+                } catch (IOException e) {
+                    Log.w(TAG, "Failed to close connection: " + e.getMessage());
+                }
+            }
+        }
+    }
+
+    private void sendLongpollChanges(List<RevisionInternal> revs) {
+        sendLongpollChanges(revs, 0);
+    }
+
+    private void sendLongpollChanges(List<RevisionInternal> revs, long since) {
+        // Ensure that the content type is application/json:
+        connection.getResHeader().add("Content-Type", CONTENT_TYPE_JSON);
+        sendResponse();
+
+        OutputStream os = connection.getResponseOutputStream();
+        try {
+            Map<String, Object> body = responseBodyForChanges(revs, since);
+            if (callbackBlock != null) {
+                byte[] data = null;
+                try {
+                    data = Manager.getObjectMapper().writeValueAsBytes(body);
+                } catch (Exception e) {
+                    Log.w(TAG, "Error serializing JSON", e);
+                }
+                os.write(data);
```

Comment:
In case `writeValueAsBytes(body)` throws exception, `data` is null. Is `os.write(null)` OK?